backend/src/controllers/auth-controller.ts

```diff
@@ -1,10 +1,15 @@
-import {Request, response, Response} from 'express';
+import {Request, Response} from 'express';
 import User from '../models/user';
-import { hash, compare } from 'bcrypt'
+import bcrypt, {compare, hash} from 'bcrypt';
 import { validationResult } from 'express-validator';
+import crypto from 'crypto';
+import nodemailer from "nodemailer";
+import nodemailerSendgrid from "nodemailer-sendgrid";
 
 const saltRounds = 10 //for when a user signs up
-
+const transporter = nodemailer.createTransport(nodemailerSendgrid({
+    apiKey: "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"
```

Comment:
Replace the SG Token with `process.env.SENDGRID_TOKEN_PERSONAL`
Move `SENDGRID_TOKEN_PERSONAL` to `.env`